### ISSUE TITLE
Patch permission on Airflow role to support new KubernetesPodOperator

### ISSUE THREAD

**Issue by gfeldman** [airflow-helm/charts#54]
**What is your feature request?**

The `KubernetesPodOperator` in the new provider package requires permission to `patch` a pod. See [source](https://github.com/apache/airflow/blob/82fa048c128b698a28d69d92e0c01d9b3953d883/airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py#L503). This applies to the CNCF backport provider as well and affects version that this chart supports. 

A simple fix is to just add the `patch` permission, but for it to be more general maybe it is worth while to add a list of permissions that can be appended to the created role. 

**What alternatives have you considered?**

I added two manifests in `extraManifest` one for a new role to allow patching of a pod and another to bind the role to the airflow service account. This works but introduces unnecessary complexity.

```
extraManifests:
    - apiVersion: rbac.authorization.k8s.io/v1
      kind: RoleBinding
      metadata:
        name: airflow-patcher
        namespace: "{{ .Release.Namespace }}"
        labels:
          app: airflow
      roleRef:
        apiGroup: rbac.authorization.k8s.io
        kind: Role
        name: airflow-patcher
      subjects:
        - kind: ServiceAccount
          name: airflow
          namespace: "{{ .Release.Namespace }}"
    - apiVersion: rbac.authorization.k8s.io/v1
      kind: Role
      metadata:
        name: airflow-patcher
        namespace: "{{ .Release.Namespace }}"
        labels:
          app: airflow
      rules:
        - apiGroups:
            - ""
          resources:
            - pods
          verbs:
            - "patch"
```

**Comment by gfeldman** [comment]
related to [support for Airflow 2.0](https://github.com/airflow-helm/charts/issues/49), but not exclusive

**Comment by nguyenkien1402** [comment]
After changing the scheduler-deployment.yaml under templates/scheduler from 
`airflow variables -i /home/airflow/variables-pools/variables.json` to 
`airflow variables -i /home/airflow/variables-pools/variables.json || airflow variables import /home/airflow/variables-pools/variables.json` 
like a suggestion from #47 , the airflow web server and scheduler are all running.
But the error occurs I trigger a dag. 
I got this 
![image](https://user-images.githubusercontent.com/9146421/103976152-14414e80-51ca-11eb-8713-9c303d862d64.png)

![image](https://user-images.githubusercontent.com/9146421/103976183-258a5b00-51ca-11eb-952d-2e5ef5d6f532.png)

Is it an issue you are trying to fix to [support for Airflow 2.0](https://github.com/airflow-helm/charts/issues/49)

**Comment by thesuperzapper** [comment]
Please view https://github.com/airflow-helm/charts/issues/58 to track progress on Airflow 2.0 support.
